Z3.fs

```diff
@@ -44,6 +44,8 @@ module Expr =
             | ASub xs -> ctx.MkSub (xs |> Seq.map az |> Seq.toArray)
             | AMul xs -> ctx.MkMul (xs |> Seq.map az |> Seq.toArray)
             | ADiv (x, y) -> ctx.MkDiv (az x, az y)
+            // TODO(CaptainHayashi): refuse AMod when reals is true.
```

Comment:
If `reals` has been set, i.e. Z3 is approximating integers with reals, I suspect `ctx.MkMod` will crash because it thinks it's trying to take the modulo of two real numbers.  I haven't checked yet, though.